features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -992,6 +992,32 @@ static nsapi_error_t mbed_lwip_socket_bind(nsapi_stack_t *stack, nsapi_socket_t
         return NSAPI_ERROR_PARAMETER;
     }
 
+#if LWIP_IPV6
+    if (IP_IS_V6(ip_addr) && !ip6_addr_isany(&ip_addr)) {
+        const ip_addr_t *local_addr = mbed_lwip_get_ipv6_addr(&lwip_netif);
```

Comment:
This isn't really sufficient - you need to check all addresses for all interfaces. At present this code isn't multiple-interface, but #5558 will make it so. Basically accessing the global `lwip_netif` from this socket call is a cheat - that won't be possible from #5558. Socket calls are not interface-specific.

And even before #5558, an interface may have multiple IPv6 addresses - there currently isn't a public API to enumerate them, so it's not obvious how an application would have found any other address than `mbed_lwip_get_ipv6_addr` returns - but there may be in future.

So loop over the `netif_list`, then over each netif's addresses, looking for a match.

You may be able collapse the code a bit with some use of generic lwIP functions - the outer if could be `if !(ip_addr_isany())`, then inside that loop over interfaces, then inside that loop the V4 and V6 checks.